### ISSUE TITLE
fpcupdeluxe and OpenBSD 7.5.

### ISSUE THREAD

**Issue by fredvs** [LongDirtyAnimAlf/fpcupdeluxe#722]
Hello.

When trying to run fpcupdeluxe last release on OpenBSD 7.5 there is that error:

![fpcup_openbsd75](https://github.com/user-attachments/assets/60f2bb4b-7d94-4bd1-b607-113e31a5d177)

But I see that in last commit there was something fixed for OpenBSD.
Do you think that compile the source of fpcupdeluxe would make it run-able on OpenBSD 7.5?
If yes how to compile it because using official fpc-main fails to compile anything.

Or maybe do you have plan to release fpcupdeluxe with last commit.

Thanks.

**Comment by LongDirtyAnimAlf** [comment]
Does the new version work for you ?
https://github.com/LongDirtyAnimAlf/fpcupdeluxe/releases/download/v2.4.0f/fpcupdeluxe-x86_64-openbsd

**Comment by fredvs** [comment]
Hello.

Sadly that new version did not work for me.

![opembsd76](https://github.com/user-attachments/assets/c85f6da0-a5e3-4725-8aae-54b1acc92ac8)

**Comment by fredvs** [comment]
Re-hello.
I did try also with https://github.com/LongDirtyAnimAlf/fpcupdeluxe/releases/download/v2.4.0f/fpcupdeluxe-x86_64-openbsd70
But also error:
 
![opembsd76_2](https://github.com/user-attachments/assets/8041de9b-3467-45c9-8d83-60b3a7d7f3e4)

**Comment by LongDirtyAnimAlf** [comment]
Please try this one.
https://github.com/LongDirtyAnimAlf/fpcupdeluxe/releases/download/v2.4.0f/fpcupdeluxe-x86_64-openbsd76

**Comment by fredvs** [comment]
> Please try this one. https://github.com/LongDirtyAnimAlf/fpcupdeluxe/releases/download/v2.4.0f/fpcupdeluxe-x86_64-openbsd76

Hello Don.
Thank you for this wonderful Christmas gift!
**fpcupdeluxe-x86_64-openbsd76** runs on my OpenBSD76 system.
But there are some errors when trying to install fpc stable.
Anyway this is a great step, we have something working, maybe I need to install something more?
![opembsd76_ok1](https://github.com/user-attachments/assets/ead3c74d-0984-41be-a378-4e207c05b967)
![opembsd76_ok2](https://github.com/user-attachments/assets/f6977a0f-0568-4096-b714-6858994c5882)
![opembsd76_ok3](https://github.com/user-attachments/assets/3f367443-ccbb-4ef9-8ab3-08770969eb6b)

**Comment by fredvs** [comment]
Re-hello.
I did try to install fpc trunk but same error:
![opembsd76_ok4](https://github.com/user-attachments/assets/136434aa-d034-4343-9bb3-5b0896a1b44b)

**Comment by LongDirtyAnimAlf** [comment]
https://gitlab.com/freepascal.org/fpc/source/-/issues/41017

**Comment by fredvs** [comment]
I suppose that you have already read this:
https://daemonforums.org/showthread.php?p=75175#post75175

**Comment by fredvs** [comment]
Re-hello Don.
I will try your workaround (I suppose it is for **/rtl/openbsd/x86_64/si_c.inc** and **/rtl/openbsd/x86_64/si_g.inc**).
`procedure c_exit(exit_code: cint); cdecl; noreturn; external name '_exit'; `

But, if it works, what could be the problems if not calling this?
`__cxa_finalize(NULL)`

**Comment by fredvs** [comment]
No, same error even with` procedure c_exit(exit_code: cint); cdecl; noreturn; external name '_exit'; `
:-(

**Comment by LongDirtyAnimAlf** [comment]
Did you disable the fetching/update of the sources when running fpcupdeluxe ?
If you make source-changes, you need to disable this "feature".
If not, your changes are overwritten.

**Comment by fredvs** [comment]
Hello Don.
Happy new year!

>Did you disable the fetching/update of the sources when running fpcupdeluxe ?

Hum, sorry but I dont see that option, where is it?
![opembsd76_ok5](https://github.com/user-attachments/assets/deb32507-8780-4b2d-94d1-477605f91914)

**Comment by fredvs** [comment]
Hello DonAlfredo.

It is sad that your patch is not yet accepted because it works perfectly to compile **LazPaint** and run it out-of-the-box.
OpenBSD is a very nice OS, it continues developing X11, has nice features, is very safe and fast...

![Image](https://github.com/user-attachments/assets/2d5619fa-3a43-44eb-8968-7b4c760a12ee)

**Comment by LongDirtyAnimAlf** [comment]
I will close this issue as things seem to have been solved with FPC trunk.
Please test with the latest release 2.4.0j.
Open a new issue if needed.
